### ISSUE TITLE
Compiling from source failed when using versions of hcc/hip from ROCm apt repository

### ISSUE THREAD

**Issue by JMadgwick** [ROCmSoftwarePlatform/tensorflow-upstream#125]
I have followed the instructions from the [readme to compile from source](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/rocm_docs/tensorflow-build-from-source.md). But the build fails.

These are the last few lines of output:
`In file included from tensorflow/core/kernels/crop_and_resize_op_gpu.cu.cc:27:
In file included from ./tensorflow/core/util/gpu_kernel_helper.h:21:
./tensorflow/core/util/gpu_device_functions.h:410:8: error: no matching function for call to '__shfl_xor'
  hi = __shfl_xor(static_cast<int>(hi), lane_mask, width);
       ^~~~~~~~~~
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/arch/HIP/hcc/Half.h:660:44: note: candidate function not viable: no known conversion from 'int' to 'Eigen::half' for 1st argument
__device__ EIGEN_STRONG_INLINE Eigen::half __shfl_xor(Eigen::half var, int laneMask, int width=warpSize) {
                                           ^
In file included from tensorflow/core/kernels/crop_and_resize_op_gpu.cu.cc:27:
In file included from ./tensorflow/core/util/gpu_kernel_helper.h:21:
./tensorflow/core/util/gpu_device_functions.h:411:8: error: no matching function for call to '__shfl_xor'
  lo = __shfl_xor(static_cast<int>(lo), lane_mask, width);
       ^~~~~~~~~~
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/arch/HIP/hcc/Half.h:660:44: note: candidate function not viable: no known conversion from 'int' to 'Eigen::half' for 1st argument
__device__ EIGEN_STRONG_INLINE Eigen::half __shfl_xor(Eigen::half var, int laneMask, int width=warpSize) {
                                           ^
1 warning and 17 errors generated.
Died at /opt/rocm/bin/hipcc line 565.
ERROR: /home/james/tensorflow/gitbuild-fromsource/tensorflow-upstream/tensorflow/core/kernels/BUILD:2132:1: output 'tensorflow/core/kernels/_objs/crop_and_resize_op_gpu/tensorflow/core/kernels/crop_and_resize_op_gpu.cu.pic.o' was not created
ERROR: /home/james/tensorflow/gitbuild-fromsource/tensorflow-upstream/tensorflow/core/kernels/BUILD:2132:1: not all outputs were created or valid
Target //tensorflow/tools/pip_package:build_pip_package failed to build
INFO: Elapsed time: 68.484s, Critical Path: 49.61s
FAILED: Build did NOT complete successfully
(venv) james@James-PC:~/tensorflow/gitbuild-fromsource/tensorflow-upstream$ `

There are hundreds of lines with similar `error: no matching function for call`. I'm not an expert and have no idea if this is the problem though.

I am running Ubuntu 16.04 and have installed the rocm repo and all the requirements. I have run the [HIP sample](https://github.com/ROCm-Developer-Tools/HIP/tree/master/samples/0_Intro/square) and it works. I have also run the [HCC sample](https://github.com/RadeonOpenCompute/hcc/wiki) and that works.

I don't understand why it doesn't build, it looks like it should be simple enough and it takes at least 5-10 mins of compiling before it get those errors and stops.

**Comment by whchung** [comment]
@deven-amd care to chime in?

**Comment by deven-amd** [comment]
@JMadgwick 
are you using the latest source? can you try to do a pull, compile again, and report if the error still persists?
thanks

**Comment by JMadgwick** [comment]
The instructions state to use:
git clone -b r1.8-rocm https://github.com/ROCmSoftwarePlatform/tensorflow-upstream.git

I also tried 1.8 and 1.7-rocm with the same results. I will try the latest master and see if that works.

**Comment by JMadgwick** [comment]
I have tried the default source (develop-upsteam), also tried some of the newest branches [here](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/branches/active) which were showing as build passing, such as develop-upstream-unit-test-fix-180813.
But none are building for me I get the same error as above for all of them.

I don't know much about this but maybe it is has something to do with the version of hcc etc. that is bundled in the repo? I will try installing the latest version of hcc which I have built from source and then try again to see if that makes any difference.

**Comment by whchung** [comment]
@JMadgwick actually could you try NOT use the latest version of hcc and hip, instead of using released binaries from stock ROCm releases. In case you choose to build hcc and hip from source, please use release branches which has been thoroughly tested:

- hcc: https://github.com/RadeonOpenCompute/hcc/tree/roc-1.8.x
- hip: https://github.com/ROCm-Developer-Tools/HIP/tree/roc-1.8.x

**Comment by JMadgwick** [comment]
I was able to get it to compile in the end. Although it takes a ridiculous amount of time. I used the latest versions of hip/hcc as I had already compiled those before I saw the comment above.
With the latest versions of hip/hcc I didn't get any complie errors. I had other problems with bazel and python venv but those relate to tensorflow itself, nothing to do with rocm.

So in conclusion the compilation error I was having didn't happen when I used the _latest hip/hcc_ instead of using the _versions installed from the ubuntu repo_ (http://repo.radeon.com/rocm/apt/debian/ xenial main).

Thanks for the help, this can be closed now as the problem doesn't seem to affect the latest versions of hcc/hip, only the older version in the repo.

In the end I wasn't actually able to use tensorflow as my R9 290 is old and [unsupported](https://github.com/ROCmSoftwarePlatform/tensorflow/issues/55), not a big deal for me as I have a modern nvidia card I can use instead. Keep up the good work with HIP, looks like it has some real potential.
